### PR TITLE
[hydrawise] Support unused zones

### LINKED ISSUE #12209: [hydrowise] Binding does not respect zone numbers, assigns in numerical order
I recently installed my Hyrowise Controller and linked it to the binding however due to some water pressure issues, I consolidated Zone 2 into Zone 1, leaving no Zone 2.

## Expected Behavior
In Openhab, Zone 2 should be blank, Zone 3 should be "Trees and Shrubs"

## Current Behavior
In Operhab, Zone 2 shows the value on "Trees and Shrubs" and there is no Zone 5 (should be Grass Beds)

## Possible Solution
Workaround is to move wires in the Controller so there are no gaps

## Steps to Reproduce (for Bugs)
Delete a zone in controller (might be easier than going in and undoing the wire)

## Your Environment
Openhab : 3.1.0
Hyrowise : 3.1.0.202106121717

``` 
2022-02-05 17:19:32.561 [TRACE] [l.api.graphql.HydrawiseGraphQLClient] - Sending Request: {"query":"{\n  me {\n    email\n    lastContact\n      controllers {\n      id\n      name\n        status {\n        summary\n        online\n        lastContact {\n          timestamp\n        }\n      }    \n      location {\n        coordinates {\n          latitude\n          longitude\n        }\n        forecast(days: 3) {\n          time\n          updateTime\n          conditions\n          averageWindSpeed {\n            value\n            unit\n          }\n          highTemperature {\n            value\n            unit\n          }\n          lowTemperature {\n            value\n            unit\n          }\n          probabilityOfPrecipitation\n          precipitation {\n            value\n            unit\n          }\n          averageHumidity\n        }\n      }\n      zones {\n        id\n        name\n        status {\n          suspendedUntil {\n            timestamp\n          }     \n        }\n        icon {\n          id\n          fileName\n          customImage {\n            id\n            url\n          }\n        }\n        number {\n          value\n          label\n        }\n        scheduledRuns {\n          summary\n          currentRun{\n             id\n            startTime {\n              timestamp\n            }\n            endTime {\n              timestamp\n            }\n            duration\n            status {\n              value\n              label\n            }\n          }\n          nextRun {\n            id\n            startTime {\n              timestamp\n            }\n            endTime {\n              timestamp\n            }\n            duration\n          }\n        }\n      }\n      sensors {\n        id\n        name\n        input {\n          number\n          label\n        }\n        status {\n          active\n          waterFlow {\n            value\n            unit\n          }\n        }\n        model {\n          modeType\n          active\n          offLevel\n          offTimer\n          delay\n        }\n      }\n    }\n  }\n}"}
2022-02-05 17:19:33.483 [TRACE] [l.api.graphql.HydrawiseGraphQLClient] - Received Response: {"data":{"me":{"email":"nicholas.gray@gmail.com","lastContact":"Fri, 07 Jan 22 23:41:25 +0000","controllers":[{"id":687834,"name":"My Controller","status":{"summary":"All good!","online":true,"lastContact":{"timestamp":1644006611}},"location":{"coordinates":{"latitude":-33.7697072,"longitude":151.2819771},"forecast":[{"time":"Sat, 05 Feb 22 00:00:00 +1100","updateTime":"Sat, 05 Feb 22 05:14:22 +1100","conditions":"Light Rain","averageWindSpeed":{"value":34,"unit":"km\/h"},"highTemperature":{"value":23,"unit":"\u00b0C"},"lowTemperature":{"value":19,"unit":"\u00b0C"},"probabilityOfPrecipitation":92.46,"precipitation":{"value":5.09,"unit":"mm"},"averageHumidity":63},{"time":"Sun, 06 Feb 22 00:00:00 +1100","updateTime":"Sat, 05 Feb 22 05:14:22 +1100","conditions":"Light Rain","averageWindSpeed":{"value":34,"unit":"km\/h"},"highTemperature":{"value":23,"unit":"\u00b0C"},"lowTemperature":{"value":19,"unit":"\u00b0C"},"probabilityOfPrecipitation":95.8,"precipitation":{"value":7.06,"unit":"mm"},"averageHumidity":65},{"time":"Mon, 07 Feb 22 00:00:00 +1100","updateTime":"Sat, 05 Feb 22 05:14:22 +1100","conditions":"Light Rain","averageWindSpeed":{"value":34,"unit":"km\/h"},"highTemperature":{"value":23,"unit":"\u00b0C"},"lowTemperature":{"value":18,"unit":"\u00b0C"},"probabilityOfPrecipitation":94.5,"precipitation":{"value":5.56,"unit":"mm"},"averageHumidity":66}]},"zones":[{"id":4473267,"name":"Bamboo","status":{"suspendedUntil":null},"icon":{"id":1,"fileName":"leaf.png","customImage":null},"number":{"value":1,"label":"Zone 1"},"scheduledRuns":{"summary":"Bamboo will next run on Sunday, 6th of February 2022, 07:00AM (Sydney Time) for 30 minutes. Normal watering cycle","currentRun":null,"nextRun":{"id":"18542867346","startTime":{"timestamp":1644091200},"endTime":{"timestamp":1644093000},"duration":30}}},{"id":4469122,"name":"Trees and Shrubs","status":{"suspendedUntil":null},"icon":{"id":4,"fileName":"flower.png","customImage":null},"number":{"value":3,"label":"Zone 3"},"scheduledRuns":{"summary":"Trees and Shrubs will next run on Sunday, 6th of February 2022, 07:30AM (Sydney Time) for 15 minutes. Normal watering cycle","currentRun":null,"nextRun":{"id":"18542867347","startTime":{"timestamp":1644093000},"endTime":{"timestamp":1644093900},"duration":15}}},{"id":4472303,"name":"Grass Pool","status":{"suspendedUntil":null},"icon":{"id":10,"fileName":"grass.png","customImage":null},"number":{"value":4,"label":"Zone 4"},"scheduledRuns":{"summary":"Grass Pool will next run on Sunday, 6th of February 2022, 07:45AM (Sydney Time) for 20 minutes. Normal watering cycle","currentRun":null,"nextRun":{"id":"18542867348","startTime":{"timestamp":1644093900},"endTime":{"timestamp":1644095100},"duration":20}}},{"id":4472467,"name":"Grass Beds","status":{"suspendedUntil":null},"icon":{"id":10,"fileName":"grass.png","customImage":null},"number":{"value":5,"label":"Zone 5"},"scheduledRuns":{"summary":"Grass Beds will next run on Sunday, 6th of February 2022, 08:05AM (Sydney Time) for 20 minutes. Normal watering cycle","currentRun":null,"nextRun":{"id":"18542867349","startTime":{"timestamp":1644095100},"endTime":{"timestamp":1644096300},"duration":20}}}],"sensors":[]}]}}}
2022-02-05 17:19:33.486 [TRACE] [l.handler.HydrawiseControllerHandler] - onData my controller id 687834
2022-02-05 17:19:33.486 [TRACE] [l.handler.HydrawiseControllerHandler] - Updating Controller 687834 sensors [] forecast [org.openhab.binding.hydrawise.internal.api.graphql.schema.Forecast@79164252, org.openhab.binding.hydrawise.internal.api.graphql.schema.Forecast@ad45d70, org.openhab.binding.hydrawise.internal.api.graphql.schema.Forecast@681d03d2] 
2022-02-05 17:19:33.487 [DEBUG] [l.handler.HydrawiseControllerHandler] - TEMP forecast1 temperaturehigh ?C 23
2022-02-05 17:19:33.487 [DEBUG] [l.handler.HydrawiseControllerHandler] - TEMP forecast1 temperaturelow ?C 19
2022-02-05 17:19:33.487 [DEBUG] [l.handler.HydrawiseControllerHandler] - TEMP forecast2 temperaturehigh ?C 23
2022-02-05 17:19:33.487 [DEBUG] [l.handler.HydrawiseControllerHandler] - TEMP forecast2 temperaturelow ?C 19
2022-02-05 17:19:33.488 [DEBUG] [l.handler.HydrawiseControllerHandler] - TEMP forecast3 temperaturehigh ?C 23
2022-02-05 17:19:33.488 [DEBUG] [l.handler.HydrawiseControllerHandler] - TEMP forecast3 temperaturelow ?C 18
2022-02-05 17:19:33.488 [TRACE] [l.handler.HydrawiseControllerHandler] - Updateing Zone zone1 Bamboo 
2022-02-05 17:19:33.489 [TRACE] [l.handler.HydrawiseControllerHandler] - Updateing Zone zone2 Trees and Shrubs 
2022-02-05 17:19:33.489 [TRACE] [l.handler.HydrawiseControllerHandler] - Updateing Zone zone3 Grass Pool 
2022-02-05 17:19:33.489 [TRACE] [l.handler.HydrawiseControllerHandler] - Updateing Zone zone4 Grass Beds 
2022-02-05 17:19:33.489 [TRACE] [l.handler.HydrawiseControllerHandler] - onData my controller id 687834
2022-02-05 17:19:33.489 [TRACE] [l.handler.HydrawiseControllerHandler] - Updating Controller 687834 sensors [] forecast [org.openhab.binding.hydrawise.internal.api.graphql.schema.Forecast@79164252, org.openhab.binding.hydrawise.internal.api.graphql.schema.Forecast@ad45d70, org.openhab.binding.hydrawise.internal.api.graphql.schema.Forecast@681d03d2] 
2022-02-05 17:19:33.489 [DEBUG] [l.handler.HydrawiseControllerHandler] - TEMP forecast1 temperaturehigh ?C 23
2022-02-05 17:19:33.490 [DEBUG] [l.handler.HydrawiseControllerHandler] - TEMP forecast1 temperaturelow ?C 19
2022-02-05 17:19:33.490 [DEBUG] [l.handler.HydrawiseControllerHandler] - TEMP forecast2 temperaturehigh ?C 23
2022-02-05 17:19:33.490 [DEBUG] [l.handler.HydrawiseControllerHandler] - TEMP forecast2 temperaturelow ?C 19
2022-02-05 17:19:33.490 [DEBUG] [l.handler.HydrawiseControllerHandler] - TEMP forecast3 temperaturehigh ?C 23
2022-02-05 17:19:33.491 [DEBUG] [l.handler.HydrawiseControllerHandler] - TEMP forecast3 temperaturelow ?C 18
2022-02-05 17:19:33.491 [TRACE] [l.handler.HydrawiseControllerHandler] - Updateing Zone zone1 Bamboo 
2022-02-05 17:19:33.491 [TRACE] [l.handler.HydrawiseControllerHandler] - Updateing Zone zone2 Trees and Shrubs 
2022-02-05 17:19:33.491 [TRACE] [l.handler.HydrawiseControllerHandler] - Updateing Zone zone3 Grass Pool 
2022-02-05 17:19:33.491 [TRACE] [l.handler.HydrawiseControllerHandler] - Updateing Zone zone4 Grass Beds 
```

### PR DESCRIPTION
**Pull request by digitaldan**
Fixes #12209

Signed-off-by: Dan Cunningham <dan@digitaldan.com>

### CODE REVIEW

File: bundles/org.openhab.binding.hydrawise/src/main/java/org/openhab/binding/hydrawise/internal/handler/HydrawiseControllerHandler.java [hunk 1]
```diff
@@ -274,9 +274,15 @@ private void updateController(Controller controller) {
...
         AtomicReference<Boolean> anySuspended = new AtomicReference<Boolean>(false);
-        int i = 1;
         for (Zone zone : zones) {
-            String group = "zone" + (i++);
+            // there are 12 relays per expander, expanders will have a zoneNumber like:
+            // 10 for expander 0, relay 10 = zone10
+            // 101 for expander 1, relay 1 = zone13
+            // 212 for expander 2, relay 12 = zone36
+            // division of integers in Java give whole numbers, not remainders FYI
+            int zoneNumber = ((zone.number.value / 100) * 12) + zone.number.value - ((zone.number.value / 100) * 100);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.hydrawise/src/main/java/org/openhab/binding/hydrawise/internal/handler/HydrawiseControllerHandler.java:283]: Suggestion:
```
             int zoneNumber = ((zone.number.value / 100) * 12) + (zone.number.value % 100);
```

**Review by lolodomo — COMMENTED** [review]
One suggestion

File: bundles/org.openhab.binding.hydrawise/src/main/java/org/openhab/binding/hydrawise/internal/handler/HydrawiseControllerHandler.java (hunk 1 above)
Comment (digitaldan) [reply bundles/org.openhab.binding.hydrawise/src/main/java/org/openhab/binding/hydrawise/internal/handler/HydrawiseControllerHandler.java:283]: Good sugestion!

**Comment by digitaldan** [comment]
Thanks!

**Review by lolodomo — APPROVED** [review]
LGTM

**Pull request merged**